reconcile/test/test_utils_terrascript_aws_client.py

```diff
@@ -49,27 +50,79 @@ def is_job_running(self, name):
 
 
 def test_use_previous_image_id_no_upstream(ts):
-    assert ts._use_previous_image_id({}) is False
+    assert ts._use_previous_image_id([]) is False
 
 
-def test_use_previous_image_id_false(mocker, ts):
-    result = False
+@pytest.mark.parametrize("result", [True, False])
+def test_use_previous_image_id(mocker, ts, result):
     mocker.patch(
         "reconcile.utils.terrascript_aws_client.TerrascriptClient.init_jenkins",
         return_value=MockJenkinsApi(result),
     )
-    image = {"upstream": {"instance": {"name": "ci"}, "name": "job"}}
+    image = [{"upstream": {"instance": {"name": "ci"}, "name": "job"}}]
     assert ts._use_previous_image_id(image) == result
 
 
-def test_use_previous_image_id_true(mocker, ts):
-    result = True
+def test_get_asg_image_id(mocker, ts: tsclient.TerrascriptClient):
+    awsapi = mocker.patch("reconcile.utils.terrascript_aws_client.AWSApi")
+    get_image_id_mock = awsapi().get_image_id
+    get_image_id_mock.return_value = "ami-123456"
+
+    ref = "sha-12345"
     mocker.patch(
-        "reconcile.utils.terrascript_aws_client.TerrascriptClient.init_jenkins",
-        return_value=MockJenkinsApi(result),
+        "reconcile.utils.terrascript_aws_client.TerrascriptClient._get_commit_sha",
```

Comment:
not a fan of mocking private methods :(

Reply:
I'm never a big fan of mocking at all, but I don't want to test `_get_commit_sha` implicitly. Do you have an idea for a better approach?

Reply:
make it public :-D

Reply:
Or maybe mocking `github.get_repo.get_commit`?